### ISSUE TITLE
[Bug Report] ttnn.sum resulting shape is wrong

### ISSUE THREAD

**Issue by rfurko-tt** [tenstorrent/tt-metal#14530]
**Describe the bug**
Wrong output shape.

**To Reproduce**
```
import torch
import ttnn

torch.set_printoptions(precision=10)

with ttnn.manage_device(device_id=0) as device:
    x_torch = torch.ones((1, 1, 9, 9), dtype=torch.float32)
    x = ttnn.from_torch(x_torch, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)

    y = ttnn.sum(x)
    y0 = ttnn.sum(x, dim=0)
    y1 = ttnn.sum(x, dim=1)
    y2 = ttnn.sum(x, dim=2)
    y3 = ttnn.sum(x, dim=3)

    print('y:', y.shape)
    print('y0:', y0.shape)
    print('y1:', y1.shape)
    print('y2:', y2.shape)
    print('y3:', y3.shape)
```

current output:
```
y: ttnn.Shape([1, 1, 1[32], 1[32]])
y0: ttnn.Shape([1, 1, 32, 32])
y1: ttnn.Shape([1, 1, 32, 32])
y2: ttnn.Shape([1, 1, 1[32], 9[32]])
y3: ttnn.Shape([1, 1, 9[32], 1[32]])
```

**Expected behavior**
```
y: ttnn.Shape([1, 1, 1[32], 1[32]])
y0: ttnn.Shape([1, 1, 9[32], 9[32]])
y1: ttnn.Shape([1, 1, 9[32], 9[32]])
y2: ttnn.Shape([1, 1, 1[32], 9[32]])
y3: ttnn.Shape([1, 1, 9[32], 1[32]])
```

**Please complete the following environment information:**
- OS: Ubuntu 20.04
- Version of software: latest main

**Comment by dmakoviichuk-tt** [comment]
hey @yugaoTT  if you know better person feel free to move.

**Comment by yugaoTT** [comment]
@yan-zaretskiy I assigned this to you as one of the code owners for eltwise. Please re-assign if needed. Thanks

**Comment by dmakoviichuk-tt** [comment]
@yugaoTT Is reduction op a part of the elementwise?

**Comment by yugaoTT** [comment]
@dmakoviichuk-tt yes

**Comment by dmakoviichuk-tt** [comment]
@davorchap it is one of the most annoying issues together with ttnn::mean.

**Comment by davorchap** [comment]
> [@davorchap](https://github.com/davorchap) it is one of the most annoying issues together with ttnn::mean.

@bbradelTT and his team are also taking on reduction Ops (sum, mean, etc)

**Comment by bbradelTT** [comment]
The AutoFormat code is causing that and needs to be removed.

**Comment by dmakoviichuk-tt** [comment]
@bbradelTT when it could be done? Can we just register functions without autoformat?

**Comment by dmakoviichuk-tt** [comment]
cc @ayerofieiev-tt  @sminakov-tt  do you have any plans related t othe autoformat?

**Comment by ayerofieiev-tt** [comment]
@dmakoviichuk-tt not anywhere soon

**Comment by bbradelTT** [comment]
Blocked by https://github.com/tenstorrent/tt-metal/issues/15511

There will be nothing to update until that bug is fixed.

**Comment by bbradelTT** [comment]
> [@bbradelTT](https://github.com/bbradelTT) when it could be done? Can we just register functions without autoformat?

@dmakoviichuk-tt The blocking issue will hopefully be fixed by end of next week. I would expect another day or two after that to make updates for this issue.

**Comment by bbradelTT** [comment]
Update:
- still blocked

**Comment by bbradelTT** [comment]
Update:
- still blocked

**Comment by rfurko-tt** [comment]
Hi @bbradelTT,
Could you please explain/add information why [#15511](https://github.com/tenstorrent/tt-metal/issues/15511) is blocking the ticket? 
It seems that the problem is missing shape, not values to pad or lack of padding.

**Comment by bbradelTT** [comment]
@rfurko-tt to get the shape set up, reduce ops call reshape, which calls pad, which doesn't work or fails in some scenarios.

**Comment by bbradelTT** [comment]
Update:
- still blocked
- expected completion of underlying issue that is blocking work on this one has been moved from this week to next week

**Comment by bbradelTT** [comment]
Update:
- still waiting for underlying issues to be fixed

**Comment by bbradelTT** [comment]
Update:
- still waiting for underlying issues to be fixed

**Comment by bbradelTT** [comment]
Update:
- still waiting for underlying issues to be fixed

**Comment by bbradelTT** [comment]
Update:
- still waiting for underlying issues to be fixed

**Comment by bbradelTT** [comment]
Update:
- still waiting for underlying issues to be fixed

**Comment by bbradelTT** [comment]
Output when running the test with latest main:
```
y: ttnn.Shape([1, 1, 1[32], 1[32]])
y0: ttnn.Shape([1, 1, 9[32], 9[32]])
y1: ttnn.Shape([1, 1, 9[32], 9[32]])
y2: ttnn.Shape([1, 1, 1[32], 9[32]])
y3: ttnn.Shape([1, 1, 9[32], 1[32]])
```

Closing.
